### PR TITLE
Fix database-cleaner config

### LINKED ISSUE #246: Issue with truncation DB cleaner, while running feature capybara specs with JS
Bumped into this issue while was testing form actions, based on AJAX with capybara feature specs.
Spec got frozen, and bin/ci couldn't pass.
Got this fixed in my PDP project - https://github.com/maximlarionov/pdp-blog/commit/032058b6acd408ddaf768c5a72bbd89f152d3bee#diff-4d41f2688fb9abaa2897572d409789f3L1

Also, found an original issue https://github.com/DatabaseCleaner/database_cleaner/issues/273

I guess, it needs some investigating, why it's happening, and how we should fix it.

### PR DESCRIPTION
**Pull request by KirillKayumov**
This fixes #246

### CODE REVIEW

**Comment by timurvafin** [comment]
@maxprokopiev could you please make sure this changes fix your issues completely.

File: spec/support/database_cleaner.rb [hunk 1]
```diff
@@ -4,11 +4,13 @@
     DatabaseCleaner.clean_with(:truncation)
```
Comment (maximlarionov) [inline spec/support/database_cleaner.rb]: well, that line isn't needed :)

before(:suite) works only once, like before(:all), but being called even before than before(:all)

before(:each) is being called before each running spec. so in your ```ruby config.before(:each)```
you reset strategy each time you check new spec.

Comment (maximlarionov) [inline spec/support/database_cleaner.rb]: well, '''DatabaseCleaner.strategy = :transaction''' isn't needed anymore :)

before(:suite) works only once, like before(:all), but being called even before than before(:all)

before(:each) is being called before each running spec. so in your config.before(:each)
you reset strategy each time you check new spec.

Comment (KirillKayumov) [inline spec/support/database_cleaner.rb]: @maximlarionov got it, thanks! Please check the last commit

Comment (maximlarionov) [inline spec/support/database_cleaner.rb]: cool :)

You may also update database cleaner gem to latest version

Gemfile
```gem 'database_cleaner', git: 'git@github.com:DatabaseCleaner/database_cleaner.git'```

@KirillKayumov

Comment (KirillKayumov) [inline spec/support/database_cleaner.rb]: @maximlarionov I don't know is it really necessary, because at least latest version doesn't fix the bug

**Comment by maximlarionov** [comment]
Well, everything else is :ok: for me. 

```config.around(:each)``` was causing troubles with hanging specs, trying to call DatabaseCleaner truncation in middle of AR transaction, so everything just hung completely.
Switching to before-after solves this problem.

cc @timurvafin

**Comment by dmitrytrager** [comment]
:ok:

File: spec/support/database_cleaner.rb (hunk 1 above)
Comment (maximlarionov) [inline spec/support/database_cleaner.rb]: Well, just to keep newest version of gem.

**Comment by timurvafin** [comment]
@KirillKayumov @maximlarionov based on this tread http://stackoverflow.com/questions/11419536/postgresql-truncation-speed/11423886#11423886 looks like DELETE will work faster for us.

Could you please try to use this config https://github.com/thoughtbot/suspenders/blob/master/templates/database_cleaner_rspec.rb

**Comment by KirillKayumov** [comment]
@timurvafin I've tried this config in my PDP project where I had freezing of feature specs. Now it works fine. In rails-base it works too.

**Comment by timurvafin** [comment]
@KirillKayumov could you please use config exactly I provided?

**Comment by KirillKayumov** [comment]
@timurvafin sure. Done

**Pull request merged**